src/Microsoft.AspNetCore.Server.Kestrel/Http/FrameOfT.cs

```diff
@@ -144,6 +135,8 @@ public class Frame<TContext> : Frame
             }
             finally
             {
+                // Error occurred, do not return components to pool
+                ResetComponents(poolingPermitted: false);
```

Comment:
Actually, I think I understand now: you're afraid of buggy apps. I think header pooling should also be opt-in if we want to support it at all.